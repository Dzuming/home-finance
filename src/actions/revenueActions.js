import * as types from './actionTypes';
import { createAction } from 'redux-actions';
import env from '../../environments/config';

export const fetchRevenue = date => (dispatch, getState) => {
  const { id } = getState().user;
  return dispatch({
    type: types.API_REQUEST_GET,
    payload: {
      url: `${env.api_url}/api/revenue/${id}/${date}`,
      success: setRevenue,
    },
  });
};

export const setRevenue = createAction(types.REVENUE_CRUD.READ);
