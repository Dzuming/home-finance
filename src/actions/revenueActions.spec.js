import * as actions from './revenueActions';
import expect from 'expect';

const revenue = {
  revenue: 50000
};

describe('revenue actions', () => {
  it('should create API_REQUEST_PENDING action with snapshot', () => {
    expect(actions.setRevenue(revenue)).toMatchSnapshot();
  });
});
