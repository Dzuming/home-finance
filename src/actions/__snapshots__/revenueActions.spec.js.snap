// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`revenue actions should create API_REQUEST_PENDING action with snapshot 1`] = `
Object {
  "payload": Object {
    "revenue": 50000,
  },
  "type": "READ_REVENUE",
}
`;
